ordering/__init__.py

```diff
@@ -103,6 +104,14 @@ def assert_new_item(self, item: T) -> None:
         if item in self:
             raise KeyError("Ordering {} already contains {}".format(self, item))
 
+    @classmethod
+    def assert_unique_items(cls, items: Collection[_T]) -> None:
+        if len(items) > len(set(items)):
+            d: Dict[_T, bool] = {}
+            duplicates = [i for i in items if i in d or d.setdefault(i, False)]
```

Comment:
Fair enough, I'm also wondering if it's worth getting *all* the duplicate items, or just returning the first duplicate, à la overlapping positional and keyword arguments, i.e.
```Python
>>> (lambda x, y: None)(1, 2, x=1, y=2)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: <lambda>() got multiple values for argument 'x'
```